Server/game.ts

```diff
@@ -1,22 +1,62 @@
+const buildings = require("../../Client/buildings.json"); // require() automatically parses the json
+const units = require("../../Client/units.json");
+
 class GameInfo {
     private turn: Team = 0;
-    private field: Tile[][] = [];
+    private field!: Field; // [0][0] is top left corner. [x] moves right, [y] moves down
+    private fieldSize: number // Not sure if this is necessary
+    private players: Player[][] = [[], []];
 
-    constructor (fieldSize=50) {
+    constructor (players: Player[], fieldSize=50) {
+        players.forEach(p=>this.players[p.team].push(p));
+        this.fieldSize = fieldSize;
         this.setField(fieldSize);
     }
 
     setField(size: number) {
         // Currently assuming square fields only
-        for (let i = 0; i < size; i++) {
-            const row: Tile[] = [];
-            for (let j = 0; j < size; j++) {
-                row.push(new Tile());
+        // Create field
+        this.field = new Field(size);
+        // Make HQs
+        // TODO: Fix later: Right now, we will simply spawn HQs in the corners
+        const stats: BuildingStats = buildings["hq"];
+        // TODO: Fix later: Right now, assumes there is enough space for the HQs
+        if (size < stats.size * 2 ) {throw new Error("field too small");}
+        for (let i = 0; i <= 1; i++) {
+            for (let j = 0; j <= 1; j++) {
+                // Team 0's HQs are at the top corners of the map, and Team 1 the bottom corners
+                this.spawnBuilding(stats, i * (size - stats.size), j * (size - stats.size), this.players[i][j]);
             }
-            this.field.push(row);
+        }
+        
+    }
+
+    // Spawns a building with its top left corner at (x, y)
+    // Not sure what to return if invalid
+    spawnBuilding(building: BuildingStats, x:number, y:number, owner: Player): Result<Building, undefined> {

```

Comment:
Simplify from `Result<Building, undefined>` to `Building | undefined`?

Reply:
Done: changed to `Building | null`.